src/contracts/deposit/integrity-checker/integrity-checker.service.ts

```diff
@@ -0,0 +1,141 @@
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
+import { RepositoryService } from 'contracts/repository';
+import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+import { BlockTag } from 'provider';
+import { DepositTree } from '../deposit-tree';
+import {
+  VerifiedDepositEvent,
+  VerifiedDepositEventsCache,
+} from '../interfaces';
+
+@Injectable()
+export class DepositIntegrityCheckerService {
+  finalizedTree = new DepositTree();
```

Comment:
Why do we name it finalized? It makes impression that it connects with block finality

Reply:
I want to highlight that  in historical front-run check (and check of front-run attempt too) we want to have 'latest' data.

Reply:
we have all the data we need
I just keep `finalized` separate from `latest`.